source/endpoint-reference/endpoint-folders.rst

```diff
@@ -0,0 +1,191 @@
+                 The root folder has an empty string as name.
+-------- ------- ---------------------------------------------------------------
+index    object  Behaves like a standard Shoji catalog containing the children
+                 of the folder keyed by URL. This catalog will contain a mixed
+                 set of folder and variables URLs. Clients should differentiate
+                 each of the objects by looking at the tuple's `type` attribute.
+                 Folders will have type "folder" while variables will contain
+                 their variables type.
+-------- ------- ---------------------------------------------------------------
+graph    list    A flat list containing the list of URLs of variables and
+                 subfolders on the given folder.
+======== ======= ===============================================================
+
+JSON example:
+
+.. language_specific::
+   --JSON
+   .. code:: json
+
+      {
+          "element": "shoji:catalog",
+          "self": "https://app.crunch.io/api/datasets/5ee0a0/variables/",
+          "size": 10000,
+          "body": {
+              "name": "Folder name"
+          },
+          "index": {
+              "https://app.crunch.io/datasets/abcdef/variables/123/": {
+                  "name": "Birth Year",
+                  "derived": false,
+                  "discarded": false,
+                  "alias": "birthyear",
+                  "type": "numeric",
+                  "id": "123",
+                  "notes": "",
+                  "description": "In what year were you born?"
+              },
+              "https://app.crunch.io/datasets/abcdef/folders/qwe/": {
+                  "name": "Subfolder name"
+              },
+              "https://app.crunch.io/datasets/abcdef/variables/456/": {
+                  "subvariables_catalog": "aad4ad/subvariables/",
+                  "name": "An Array",
+                  "derived": true,
+                  "discarded": false,
+                  "alias": "arrayvar",
+                  "subvariables": [
+                      "439dcf/",
+                      "1c99ea/"
+                  ],
+                  "notes": "All variable types can have notes",
+                  "type": "categorical_array",
+                  "id": "456",
+                  "description": ""
+              }
+          },
+          "graph": [
+              "https://app.crunch.io/datasets/abcdef/variables/456/",
+              "https://app.crunch.io/datasets/abcdef/variables/123/",
+              "https://app.crunch.io/datasets/abcdef/folders/qwe/"
+          ]
+      }
+
+
+Creating folders
+----------------
+
+To create a subfolder, clients must POST to the target folder (which will be the
+parent of the new subfolder).
+
+The payload must contain a `body` member which indicates only the name of
+the subfolder. Note that subfolder names must be unique between them and
+variables included.
```

Comment:
This requirement is not specified in https://github.com/Crunch-io/design/pull/9/files -- I can guess the rationale for not allowing a duplicate name between folders is to reduce user confusion, but is that also valid between folders and variables? I could easily imagine a scenario where a user has a variable "foo" and wants to 1) create a new folder "foo", then 2) move the variable "foo" into the new folder in that order, without the system throwing an error on step 1.

Reply:
I recommend deferring enforcing a name uniqueness policy, if possible.

Reply:
right now we enforce this logic when creating hierarchical orders https://github.com/Crunch-io/backend-http/blob/master/cr/server/api/validators/order.py#L86 

We don't allow two things to have the same name under a group. This follows the _Just like a file system_ moto that brought us here.

Reply:
If you're going to keep that policy then you need to add it as a requirement to https://github.com/Crunch-io/design/pull/9/files

Reply:
https://github.com/Crunch-io/design/commit/140187493cc0e3fdb6b2cbdd7b68749f808ee470